cmd/generateprofile.go

```diff
@@ -0,0 +1,680 @@
+object in the configuration path)
+
+[user|account|operator]Name - includes the friendly name for the for 
+user, account, operator, If no prefix (user/account/operator is provided, 
+it targets the last object in the configuration path)
+		`,
+
+		Args: MaxArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			us := args[0]
+			u, err := ParseNscURL(us)
+			if err != nil {
+				return fmt.Errorf("error parsing %q:%v", us, err)
+			}
+			params.nscu = u
+			q, err := u.query()
+			if err != nil {
+				return fmt.Errorf("error parsing query %q:%v", us, err)
+			}
+			if len(q) > 0 {
+				config := GetConfig()
+				oldSR := config.StoreRoot
+				oldOp := config.Operator
+				oldAc := config.Account
+				v, ok := q["store"]
+				if ok {
+					defer func() {
+						_ = config.setStoreRoot(oldSR)
+						if oldOp != "" {
+							_ = config.SetOperator(oldOp)
+						}
+						if oldAc != "" {
+							_ = config.SetAccount(oldAc)
+						}
+					}()
+					sr, err := Expand(v)
+					if err != nil {
+						return err
+					}
+					if err := config.setStoreRoot(sr); err != nil {
+						return err
+					}
+					if err := config.SetOperator(u.operator); err != nil {
+						return err
+					}
+				}
+
+				storeDir, ok := q[keystoreDir]
+				if ok {
+					ks, err := Expand(storeDir)
+					if err != nil {
+						return err
+					}
+					if err := os.Setenv(store.NKeysPathEnv, ks); err != nil {
+						return err
+					}
+				}
+			}
+			return RunAction(cmd, args, &params)
+		},
+		Hidden: true,
+	}
+	cmd.Flags().StringVarP(&params.outputFile, "output-file", "o", "--", "output file, '--' is stdout")
+
+	return cmd
+}
+
+func init() {
+	generateCmd.AddCommand(createProfileCmd())
+}
+
+type Arg string
+
+const (
+	prefix           = "nsc://"
+	operatorKey  Arg = "operatorkey"
+	accountKey   Arg = "accountkey"
+	userKey      Arg = "userkey"
+	key          Arg = "key"
+	operatorSeed Arg = "operatorseed"
+	accountSeed  Arg = "accountseed"
+	userSeed     Arg = "userseed"
+	seed         Arg = "seed"
+	operatorName Arg = "operatorname"
+	accountName  Arg = "accountname"
+	userName     Arg = "username"
+	name         Arg = "name"
+	keystoreDir  Arg = "keystore"
+	storeDir     Arg = "store"
+)
+
+type ProfileCmdParams struct {
+	nscu       *NscURL
+	results    *Profile
+	oc         *jwt.OperatorClaims
+	ac         *jwt.AccountClaims
+	uc         *jwt.UserClaims
+	outputFile string
+}
+
+type Details struct {
+	Service []string `json:"service,omitempty"`
+	Name    string   `json:"name,omitempty"`
+	Seed    string   `json:"seed,omitempty"`
+	Key     string   `json:"id,omitempty"`
+}
+
+type Profile struct {
+	UserCreds string   `json:"user_creds,omitempty"`
+	Operator  *Details `json:"operator,omitempty"`
+	Account   *Details `json:"account,omitempty"`
+	User      *Details `json:"user,omitempty"`
+}
+
+type NscURL struct {
+	operator string
+	account  string
+	user     string
+	qs       string
+}
+
+type stringSet struct {
+	set map[string]string
+}
```

Comment:
type stringSet map[string]struct{} would have worked as well. Presence is the only thing checked.